### PR TITLE
Fix #124 #196 - adds a copy button to the share menu to copy short URL

### LINKED ISSUE #124: add copy link to top share menu
copies short link to clipboard with css popup

### PR DESCRIPTION
**Pull request by prebhdevsingh**

### CODE REVIEW

File: src/js/renderControls.js [hunk 1]
```diff
@@ -130,3 +138,33 @@ function renderControls() {
     </div>
   );
 }
+
+function copyShortUrl() {
```
[the remark this answers is not in the document]
Comment (bogas04) [reply src/js/renderControls.js]:
```js
const copyToClipboard = text => new Promise((resolve, reject) => {
  try {
    const textarea = document.createElement('textarea');
    textarea.textContent = text;
    document.body.appendChild(textarea);

    const selection = document.getSelection();
    const range = document.createRange();
    range.selectNode(textarea);
    selection.removeAllRanges();
    selection.addRange(range);

    const result = document.execCommand('copy');

    selection.removeAllRanges();
    document.body.removeChild(textarea);
    
    if (result) {
      resolve();
    } else {
      throw new Error('Failed to copy');
    }
  } catch (e) {
    reject(e);
  }
});
```

[the remark this answers is not in the document]
Comment (prebhdevsingh) [reply src/js/renderControls.js]: I have added the utility function to main.js. I hope this is the right place for it.

[the remark this answers is not in the document]
Comment (prebhdevsingh) [reply src/js/renderControls.js]: The above code doesn't work in IE Edge

File: src/js/renderControls.js [hunk 2]
```diff
@@ -81,8 +81,15 @@ function renderControls() {
               <span class="sr-only">Share via email</span>
             </a>
           </li>
+          <li>
+            <a id="copy-short-url" class="copy" onclick={"copyShortUrl()"} >
```
Comment (bogas04) [inline src/js/renderControls.js]: While this will totally work, can you please use `click={copyShortUrl}` instead?

**Comment by prebhdevsingh** [comment]
Yes, tested!

File: src/js/main.js [hunk 3]
```diff
@@ -242,30 +242,6 @@ function displayOptionToggle(e) {
   $display.classList.toggle(option);
 
   checkboxPref(e, 'displayOptions', option);
-
-  // Update the textarea for copy/social sharing
-  [...document.querySelectorAll('.display .line')]
```
Comment (bogas04) [inline src/js/main.js]: This was anyway very ugly method of doing this.

File: src/js/main.js [hunk 4]
```diff
@@ -287,3 +263,41 @@ $search.onkeyup = function () {
...
+    } else {
+      throw new Error('Failed to copy');
+    }
+  } catch (e) {
+    reject(e);
+  }
+});
+
+const showToast = (text, delay = 2500) => new Promise(resolve => {
+  const copyURLconfirm = document.getElementById('toast-notification');
```
Comment (bogas04) [inline src/js/main.js]: This is more generic method.

**Comment by bogas04** [comment]
@navdeepsinghkhalsa & @prebhdevsingh, could you please review & test it once? Thanks!

File: src/js/renderControls.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 function renderControls() {
   return (
-    <div id="controls-wrapper" class="no-select hidden">
+   <div id="controls-wrapper" class="no-select hidden">
```
Comment (navdeepsinghkhalsa) [inline src/js/renderControls.js]: Minor indenting error. Will approve if this is the only thing.

**Comment by navdeepsinghkhalsa** [comment]
Please feel free to merge when ready

**Comment by bogas04** [comment]
There's a minor bug with copying of line. Will take it in next sprint or so.

**Pull request merged**